src/main/java/com/lists/CircularLinkedList.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * File: CircularLinkedList.java
+ * Author: Prakhar Shrivastava
+ * Date: 27 Aug, 2016
+ * -----------------------------------------
+ * CircularLinkedList is a generic data structure to store data items
+ * of any type in a continuous list fashion.
+ */
+package com.lists;
+
+
+import java.util.Iterator;
+
+/*
+* LinkedList provides following API
+        * ----------------------------------------------
+        *         LinkedList ()                   // constructs a empty list. O(1)
+        * int     getSize()                      // returns the length of list in integer. O(1)
+        * boolean isEmpty()               // returns true if list is empty, false otherwise. O(1)
+        * void    add(Item item)      	   // append an item in the last of list. O(1)
+        * void    add(Item item, int index)  //append an item at the given index in the list. O(n)
+        * Item    get(int index)             // returns the item at the given index from the list. O(n)
+        * boolean remove(int index)	   // remove the node and return true at the given index from the list. O(n)
+        * Iterator<Item> iterator()   	   // Iterate over the linkedList
+        *
+*/
+
+public class CircularLinkedList<Item> implements Iterable<Item>{
+    LinkedListNode head, tail;
+    int size;
+
+    /*
```

Comment:
Good attempt for commenting, but for public functions use java style commenting /**  */. Because we have a way to extract java doc which is very useful when we just want to browse the API in whole.